src/pymor/algorithms/newton.py

```diff
@@ -102,60 +110,103 @@ def newton(operator, rhs, initial_guess=None, mu=None, error_product=None, least
     U = initial_guess.copy()
     residual = rhs - operator.apply(U, mu=mu)
 
-    err = residual.norm(error_product)[0]
-    logger.info(f'      Initial Residual: {err:5e}')
+    # compute norms
+    solution_norm = U.norm(error_product)[0]
+    solution_norms = [solution_norm]
+    update_norms = []
+    residual_norm = residual.norm(error_product)[0]
+    residual_norms = [residual_norm]
+
+    # select error measure for convergence criteria
+    err = residual_norm if error_measure == 'residual' else solution_norm
```

Comment:
If the initial guess is the 0 element, then `solution_norm` is already 0. Hence, when `atol` is greater than 0, the algorithm will stop during the first iteration due to reaching the prescribed absolute tolerance, I think.

Reply:
Right, I was only thinking about `rtol`. Should probably be initialized with `inf`.

Reply:
Yes, `inf` should work.

Reply:
Fixed.